source/common/http/async_client_impl.cc

```diff
@@ -39,6 +39,7 @@ AsyncClient::Request* AsyncClientImpl::send(MessagePtr&& request, AsyncClient::C
                                             const Optional<std::chrono::milliseconds>& timeout) {
   AsyncRequestImpl* async_request =
       new AsyncRequestImpl(std::move(request), *this, callbacks, timeout);
+  async_request->initialize();
```

Comment:
Do we want consistency in `AsyncStreamImpl` with the initialization pattern? I.e. it might be surprising to someone using both `AsyncRequestImpl` and `AsyncStreamImpl` that they have slightly different ways to kick off a request after construction.